sonic-gnmi-standalone/API_REFERENCE.md

```diff
@@ -0,0 +1,302 @@
+# SONiC Upgrade Service API Reference
+
+This document provides comprehensive API documentation for the SONiC Upgrade Service gRPC interface.
+
+## Overview
+
+The SONiC Upgrade Service provides two main gRPC services:
+
+1. **SystemInfo Service** - Platform detection and system monitoring
+2. **FirmwareManagement Service** - Firmware download, cleanup, and image operations
+
+## Connection Details
+
+- **Default Port**: 50051
+- **Protocol**: gRPC over HTTP/2
+- **TLS**: Optional (configurable via environment variables)
+- **Reflection**: Enabled (supports grpcurl and other gRPC tools)
+
+## SystemInfo Service
+
+### GetPlatform
+
+Retrieves platform information for the current system.
+
+**Request**: `GetPlatformRequest` (empty)
+
+**Response**: `GetPlatformResponse`
+```protobuf
+message GetPlatformResponse {
+  string platform = 1;  // Platform identifier (e.g., "x86_64-dellemc_s5232f_c3538-r0")
+}
+```
+
+**Usage**:
+```bash
+grpcurl -plaintext localhost:50051 SystemInfo/GetPlatform
+```
+
+### GetDiskSpace
+
+Retrieves disk space information for specified paths.
+
+**Request**: `GetDiskSpaceRequest`
+```protobuf
+message GetDiskSpaceRequest {
+  repeated string paths = 1;  // Paths to check disk space for
+}
+```
+
+**Response**: `GetDiskSpaceResponse`
+```protobuf
+message GetDiskSpaceResponse {
+  repeated DiskSpaceInfo disk_spaces = 1;
+}
+
+message DiskSpaceInfo {
+  string path = 1;           // The path that was checked
+  uint64 total_bytes = 2;    // Total disk space in bytes
+  uint64 available_bytes = 3; // Available disk space in bytes
+  uint64 used_bytes = 4;     // Used disk space in bytes
+}
+```
+
+**Usage**:
+```bash
+grpcurl -plaintext -d '{"paths": ["/", "/tmp"]}' localhost:50051 SystemInfo/GetDiskSpace
+```
+
+## FirmwareManagement Service
+
+### DownloadFirmware
+
+Downloads firmware from a URL with MD5 checksum validation and optional network interface binding.
```

Comment:
If the destination path already exists on the device, will this API redownload it or reuse the existing file?